pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1918,6 +1918,9 @@ private void createPendingLoadTopic() {
     }
 
     public CompletableFuture<PartitionedTopicMetadata> fetchPartitionedTopicMetadataCheckAllowAutoCreationAsync(TopicName topicName) {
+        if(pulsar.getNamespaceService() == null){
+            return FutureUtil.failedFuture(new NullPointerException("namespaceService is null"));
```

Comment:
I think we should return an explicit error to tell the application that the namespace service is not ready.